### ISSUE TITLE
 🤖 Baseline ML Models

### ISSUE THREAD

**Issue by CSFelix** [CSFelix/Housing-Price-Competition#8]
* [x] Random Forest Regressor
* [x] XGBoost
* [x] CatBoost

**Comment by CSFelix** [comment]
> Now, even though RNNs are quite powerful, they suffer from **Vanishing Gradient Problem** which hinders them from using long term information, like they are good for storing memory 3-4 instances of past iterations but larger number of instances don't provide good results so we don't just use regular RNNs. Instead, we use a better variation of RNNs: **Long Short Term Networks (LSTM)**.

**Comment by CSFelix** [comment]
Components of LSTMs

- Forget Gate “f” ( a neural network with sigmoid);
- Candidate layer “C"(a NN with Tanh);
- Input Gate “I” ( a NN with sigmoid );
- Output Gate “O”( a NN with sigmoid);
- Hidden state “H” ( a vector );
- Memory state “C” ( a vector);

- Inputs to the LSTM cell at any step are Xt (current input) , Ht-1 (previous hidden state ) and Ct-1 (previous memory state);
- Outputs from the LSTM cell are Ht (current hidden state ) and Ct (current memory state);

**Comment by CSFelix** [comment]
* [ ] Random Forest Regressor
* [ ] XGBoost
* [ ] LSMT
* [ ] GRU
